src/metabase/models/dashboard.clj

```diff
@@ -97,11 +97,14 @@
 
 (t2/define-before-update :model/Dashboard
   [dashboard]
-  (u/prog1 (maybe-populate-initially-published-at dashboard)
-    (params/assert-valid-parameters dashboard)
-    (parameter-card/upsert-or-delete-from-parameters! "dashboard" (:id dashboard) (:parameters dashboard))
-    (collection/check-collection-namespace Dashboard (:collection_id dashboard)))
-  (maybe-populate-initially-published-at dashboard))
+  (let [changes (t2/changes dashboard)]
+    (u/prog1 (maybe-populate-initially-published-at dashboard)
+     (params/assert-valid-parameters dashboard)
+     (parameter-card/upsert-or-delete-from-parameters! "dashboard" (:id dashboard) (:parameters dashboard))
+     (collection/check-collection-namespace Dashboard (:collection_id dashboard))
+     (when (and (contains? changes :archived)
+                (:archived changes))
```

Comment:
Why do we need the first check? Surely the value can only be truthy if it's present - how exotic is this map? 😄

Reply:
Yeah, with you on that. Surely `changes` only includes the changes already? Assuming it's not some funky map with inconsistent `contains?` (filters unchanged) and `get` (returns changed and unchanged).

Reply:
yea, it's a promise from toucan2 https://github.com/camsaul/toucan2/blob/3cb36674ca0ab2e732c5ba2983db5052cf4c9b7b/src/toucan2/protocols.clj#L44

> Get a map with any changes made to `instance` since it came out of the DB. Only includes keys that have been
    added or given different values; keys that were removed are not counted. Returns `nil` if there are no changes.

Reply:
I feel like we're missing each other here - I checked the definition of `changes*` and yeah it looks like it might be a custom map, but even if so it will have completely forgotten any unchanged values. So this `contains?` check here is not necessary.

Reply:
ah you're right! my brain was somewhere for a bit.

Updated to remove the key check.